src/libraries/Directory.Build.props

```diff
@@ -52,11 +52,13 @@
     <Nullable Condition="'$(Nullable)' == '' and '$(Language)' == 'C#'">annotations</Nullable>
   </PropertyGroup>
 
-  <!-- Unique assembly versions increases(3x) the compiler throughput during reference package updates. -->
   <PropertyGroup Condition="'$(IsGeneratorProject)' == 'true'">
+    <!-- Unique assembly versions increases(3x) the compiler throughput during reference package updates. -->
     <AutoGenerateAssemblyVersion>true</AutoGenerateAssemblyVersion>
     <!-- To suppress warnings about resetting the assembly version.-->
     <AssemblyVersion />
+    <!-- Enforce extended rules around API usages in analyzers and generators to ensure our generators follow best practices. -->
+    <EnforceExtendedAnalyzerRules>true</EnforceExtendedAnalyzerRules>
```

Comment:
For other reviewing this, here's the "documentation" that attempts to explain this setting. I don't see this officially anywhere on docs.microsoft.com though: https://github.com/dotnet/roslyn-analyzers/blob/03344a064a80c1f66e50b67e3abe7335cb8ba072/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs1036-specify-analyzer-banned-api-enforcement-setting